nntrainer/graph/network_graph.cpp

```diff
@@ -353,36 +323,34 @@ void NetworkGraph::setOutputLayers() {
 
   size_t last_layer_count = 0;
   for (unsigned int idx = 0; idx < graph.size(); ++idx) {
-    auto &layer_idx = LNODE(node_list[idx])->getObject();
+    auto layer_idx = LNODE(node_list[idx]);
     for (unsigned int i = 0; i < graph.size(); ++i) {
-      auto &layer_i = LNODE(node_list[i])->getObject();
+      auto layer_i = LNODE(node_list[i]);
       if (istrequal(layer_i->getName(), layer_idx->getName()))
         continue;
-      for (unsigned int j = 0; j < layer_i->input_layers.size(); ++j) {
-        if (istrequal(layer_i->input_layers[j], layer_idx->getName())) {
+      for (unsigned int j = 0; j < layer_i->getNumInputs(); ++j) {
+        if (istrequal(layer_i->getInputLayers()[j], layer_idx->getName())) {
           bool already_exist = false;
-          for (unsigned int k = 0; k < layer_idx->output_layers.size(); ++k) {
-            if (istrequal(layer_idx->output_layers[k], layer_i->getName())) {
+          for (unsigned int k = 0; k < layer_idx->getNumOutputs(); ++k) {
+            if (istrequal(layer_idx->getOutputLayers()[k],
+                          layer_i->getName())) {
               already_exist = true;
               break;
             }
           }
 
           if (!already_exist)
-            layer_idx->output_layers.push_back(layer_i->getName());
+            layer_idx->addOutputLayers(layer_i->getName());
         }
       }
     }
 
-    if (layer_idx->getNumOutputs() != layer_idx->output_layers.size()) {
-      if (layer_idx->output_layers.size() == 0) {
+    if (layer_idx->getObject()->getNumOutputs() != layer_idx->getNumOutputs()) {
```

Comment:
Could you check this? 

```suggestion
    if (layer_idx->getNumOutputs() != layer_idx->getNumOutputs()) {
```

Reply:
no, this is intended. `nntrainer::Layer` gives `getNumOutputs()` initialized with 1. input_dims/output_dims are always assumed to be at least 1. However, `nntrainer::LayerNode` relies on size of `input_layers/output_layers` to retrieve `getNumInputs/getNumOutputs` which can be zero. (when the layer is input/loss respectively).
This is a bit confusing for now but will be united once everything moves to layer context.

Reply:
Ah, I see. Thanks for the explanation